### PR TITLE
Separate addon and test app with into Monorepo

### PR DESCRIPTION
**Pull request by vstefanovic97**

### CODE REVIEW

File: test-app/package.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
+{
+  "name": "test-app",
+  "version": "7.2.0",
```
Comment (SergeAstapov) [inline test-app/package.json:3]:
```suggestion
  "version": "7.2.0",
  "private": true,
```

**Review by SergeAstapov — CHANGES_REQUESTED** [review]
this is great first step toward v2 addon conversion!

File: test-app/package.json (hunk 1 above)
Comment (vstefanovic97) [reply test-app/package.json:3]: I added the suggestion @SergeAstapov, thank you for review!

**Review by SergeAstapov — APPROVED** [review]

**Comment by SergeAstapov** [comment]
looks great! I'll let @elwayman02 weigh in if any concerns, otherwise merge by EOD tomorrow.

File: addon/.eslintrc.js [hunk 2]
```diff
@@ -38,7 +38,6 @@ module.exports = {
         './testem.js',
         './blueprints/*/index.js',
         './config/**/*.js',
-        './tests/dummy/config/**/*.js',
```
Comment (elwayman02) [inline addon/.eslintrc.js:41]: Does this file still need the "test fles" section on lines 51-58?

File: package.json [hunk 3]
```diff
@@ -1,139 +1,30 @@
...
-  "ember": {
-    "edition": "octane"
-  },
-  "ember-addon": {
-    "configPath": "tests/dummy/config",
-    "after": [
-      "ember-sinon"
-    ]
+    "concurrently": "^8.2.1",
+    "@release-it-plugins/lerna-changelog": "^6.0.0"
```
Comment (elwayman02) [inline package.json:27]: What's the reasoning to split up this plugin dependency from the release-it dependency?

File: addon/package.json [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  },
+  "engines": {
+    "node": "14.* || 16.* || >= 18"
+  },
+  "ember": {
+    "edition": "octane"
+  },
+  "ember-addon": {
+    "after": [
+      "ember-sinon"
```
Comment (elwayman02) [inline addon/package.json:75]: Can you go ahead and drop this? We should have dropped it before, but we no longer rely on ember-sinon so this block is useless.

File: test-app/config/ember-cli-update.json [hunk 5]
```diff
@@ -6,8 +6,8 @@
       "version": "4.11.0",
       "blueprints": [
         {
-          "name": "addon",
-          "outputRepo": "https://github.com/ember-cli/ember-addon-output",
+          "name": "app",
```
Comment (elwayman02) [inline test-app/config/ember-cli-update.json:9]: Hmm, I'm not sure about this change. How exactly does ember-cli-update work with v2 addons? We still want to be able to apply the v2 addon blueprint with this config and have it affect both the `addon` and `test-app` directories.

File: addon/.eslintrc.js (hunk 2 above)
Comment (SergeAstapov) [reply addon/.eslintrc.js:41]: @elwayman02 we indeed do not need. Also there bunch of other things that could be cleaned up, but my thought process is to get this land with minimal actual changes in file and focus on moving files into locations where they should be.

as next step, we'll sync `test-app` with latest ember-cli app blueprint and sync `addon` folder setup with https://github.com/embroider-build/addon-blueprint after we conver the actual addon to v2 format.

Sorry didn't communicate this plan before, hope this makes sense as with such approach we should have more or less easily reviewable PRs.

File: package.json (hunk 3 above)
Comment (SergeAstapov) [reply package.json:27]: ah good catch! We indeed need to have `release-it` in root level package.json in `devDependencies`.

File: test-app/config/ember-cli-update.json (hunk 5 above)
Comment (SergeAstapov) [reply test-app/config/ember-cli-update.json:9]: @elwayman02 this is actually correct change.
this file will be used only for test app so we could do `npx ember-cli-update` in the `test-app` folder.

for the addon and everything, there will be separate `config/ember-cli-update.json` file which we would be able to use to run `npx ember-cli-update` in the root folder (you could see the template for that file in https://github.com/embroider-build/addon-blueprint/blob/main/files/config/ember-cli-update.json)

File: addon/package.json (hunk 4 above)
Comment (elwayman02) [reply addon/package.json:75]: It's also ok for this to be a follow-up, just wanted to call it out while I noticed it.

File: test-app/config/ember-cli-update.json (hunk 5 above)
Comment (elwayman02) [reply test-app/config/ember-cli-update.json:9]: Ok, so this file is fine but we need to add a new `config/ember-cli-update.json` as well then, right?

File: addon/.eslintrc.js (hunk 2 above)
Comment (elwayman02) [reply addon/.eslintrc.js:41]: Yup, makes sense!

File: test-app/config/ember-cli-update.json (hunk 5 above)
Comment (SergeAstapov) [reply test-app/config/ember-cli-update.json:9]: Yeap, but I would do in the separate PR as it would not make any sense just yet

Comment (elwayman02) [reply test-app/config/ember-cli-update.json:9]: Yup, seems fine!

**Comment by elwayman02** [comment]
Let's fix the release-it dependency and the other 3 comments can be follow-up PR(s). This looks great, thanks for putting in the hard work!

File: package.json (hunk 3 above)
Comment (vstefanovic97) [reply package.json]: @SergeAstapov, @elwayman02 my mistake, it should be fixed now

File: addon/package.json (hunk 4 above)
Comment (vstefanovic97) [reply addon/package.json]: @elwayman02 I removed it

**Pull request merged**